src/widgets/command_prompt.rs

```diff
@@ -1,62 +1,82 @@
+//! Command prompt for xi-term. currently this is
+//! heavily inspired by vim and is just designed to
+//! get a simple base to work off of.
+
 use std::io::Error;
 use std::io::Write;
 use termion::event::{Event, Key};
 
-use core::Command;
+use core::{Command, ParseCommandError};
 use termion::clear::CurrentLine as ClearLine;
 use termion::cursor::Goto;
 
-/// Command prompt for xi-term.
-/// currently this is heavily inspired by vim
-/// and is just disigned to get a simple base to work off of.
+use std::str::FromStr;
+
 #[derive(Debug, Default)]
 pub struct CommandPrompt {
+    dex: usize,
     chars: String,
 }
 
 impl CommandPrompt {
     /// Process a terminal event for the command prompt.
-    pub fn handle_input(&mut self, input: &Event) -> Option<Command> {
-        // TODO: not ignore arrow keys
+    pub fn handle_input(&mut self, input: &Event) -> Result<Option<Command>, ParseCommandError> {
         match input {
             Event::Key(Key::Char('\n')) => self.finalize(),
-            Event::Key(Key::Backspace) => self.back(),
-            Event::Key(Key::Char(chr)) => self.new_key(*chr),
-            _ => None,
+            Event::Key(Key::Backspace) => Ok(self.back()),
+            Event::Key(Key::Delete) => Ok(self.delete()),
+            Event::Key(Key::Left) => Ok(self.left()),
+            Event::Key(Key::Right) => Ok(self.right()),
+            Event::Key(Key::Char(chr)) => Ok(self.new_key(*chr)),
+            _ => Ok(None),
+        }
+    }
+
+    fn left(&mut self) -> Option<Command> {
```

Comment:
Do these have to return something? Since it always returns `None`, could we simplify this into:

```rust
fn left(&mut self) {
    if self.dex > 0 {
        self.dex -= 1;
    }
}
```

Reply:
The main reason is to clean up the code in CommandPrompt::handle_input, since we can then just call the function on the match arm. But it could work either way